server/app/routes.py

```diff
@@ -13,3 +20,44 @@ def videos_raw():
 	@app.route('/videos')
 	def videos():
 		return jsonify(rss.summarize_watchlist(app.config['watchlist']))
+
+	@app.route('/watchlist', methods=['POST', 'DELETE'])
+	async def modify_watchlist():
+		# only required value is 'id'
+		json = request.get_json()
+		if 'id' not in json and 'url' not in json:
+			return '', 400 # bad request
+		if 'id' in json and 'url' in json:
+			return '', 400 # bad request
+
+		args = {}
+		if 'id' in json:
```

Comment:
is there a way to check validity of ids as well? i.e. whether a channel with that id exists

Reply:
yes, however checking requires making a connection. When using channel names in the url, the id should be guaranteed to be correct since it was fetched from Youtube. However when if the id is in the url, then we only check if the id is formatted correctly.

Reply:
Note: the overhead of fetching from youtube is not very fast.

Reply:
Is it ok if the validity check is to be done only when the new channel data is fetched? #32

Reply:
Actually, the 2 tickets: this and #32 , seem to be closely related so if we end up adding id verification let's do it in a separate PR. This one is big enough.